webpack/assets/javascripts/react_app/common/MountingService.js

```diff
@@ -33,12 +33,19 @@ function mountNode(component, reactNode, data, flattenData) {
  * This is a html tag (Web component) that can be used for mounting react component from ComponentRegistry.
  */
 class ReactComponentElement extends HTMLElement {
+  static get observedAttributes() {
+    return ['data-props'];
+  }
+
   get componentName() {
     return this.getAttribute('name');
   }
-  get props() {
+  get reactProps() {
     return this.dataset.props !== '' ? JSON.parse(this.dataset.props) : {};
   }
+  set reactProps(newProps) {
+    this.dataset.props = JSON.stringify(newProps);
```

Comment:
Does this trigger a re-render as well? i.e. does webcomponents know that when `this.dataset.props` it means `data-props` attribute changed?

Reply:
yes :) It has been one of an early issues they fixed and IIUC it also got into the polyfill we are using, so it should work even in IE 10 :tada: